proposals/collection-expression-arguments.md

```diff
@@ -526,11 +528,14 @@ IReadOnlyDictionary<string, int> d = [with(comparer: StringComparer.Ordinal), ..
 
 If so, which method signatures are used when binding the arguments?
 
-For `ICollection<T>` and `IList<T>` should we use the accessible constructors from `List<T>`, or specific signatures independent from `List<T>`, say `new()` and `new(int capacity)`?
-
-For `IDictionary<TKey, TValue>` should we use the accessible constructors from `Dictionary<TKey, TValue>`, or specific signatures, say `new()`, `new(int capacity)`, `new(IEqualityComparer<K> comparer)`, and `new(int capacity, IEqualityComparer<K> comparer)`?
+Options include:
+1. Use the accessible constructors from well-known types in each case, for instance using the constructors from `List<T>` for `ICollection<T>` and `IList<T>`.
+1. Use specific signatures independent of specific type, for instance using `new()` and `new(int capacity)` for `ICollection<T>` and `IList<T>` (see [*Construction*](#construction) for all interfaces).
```

Comment:
i would separate this into two questions.  One for mutable collections and one for non-mutable collections.

Reply:
For mutable interfaces i would say the options are:

1. Use the accessible constructors for the well-known type (`List<>` or `Dictionary<,>`) guaranteed to be created foe the mutable interface type.
1. Use specific signatures independent of specific type, for instance using `new()` and `new(int capacity)` for `ICollection<T>` and `IList<T>`, and `new(IEqualityComparer<> comparer)` for `IDictionary<>` (see [*Construction*](#construction) for all interfaces).  Full set of constructors TBD.

Reply:
Won't the same options apply for both mutable and immutable interfaces?

Reply:
For non-mutable interfaces i would say the options are:

1. Use the accessible constructors from certain well-known types (like `List<>` or `Dictionary<,>`)
1. Use specific signatures for non-mutable interfaces.  Currently, the only requested specific is `with(IEqualityComparer<> comparer)` for `IReadOnlyDictionary<,>`.  Possibility of `with(int capacity)` to define an up front capacity when making sequences from inputs we don't know the length of, like `[with(capacity: 64), .. GetUnknownLengthItems1(), .. GetUnknownLengthItems2()]`.

Reply:
> Won't the same options apply for both mutable and immutable interfaces?

No.  For mutable, it's not `from well-known types in each case` in an abstract sense.  It is *concretely*: the exact set from these types.

For non-mutable, it's up for debate which well known types we'd even draw from.  It's not at all clear to me that the well known types would be `List<>` or `Dictionary<>` (i personally don't think it should be).